docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
-
-  see fpc-lifecycle-v2.puml
-end ref
-|||
-
-
-group enclave creation and attestation generation
-  
-  activate Admin
-
-  Admin -> Manager1 ++: create (CC_NAME, CC_VER, CH_name)
-  note right
-    CC_NAME: chaincode name
-    CC_VER: chaincode version
-    CH_NAME: channel name (needed by the ECC_Manager to cc2cc ECC
-  end note
-  note over Manager1 #FFaa00
+group enclave creation
+  note over Admin #FFaa00
     **MST**: we might want to replace (here and elsewhere) CC_VER with the sequence
     number from chain code definition as versions can be re-used in downgrades yet we
     might or might not want to re-use enclaves?
   end note
 
-  Manager1 -> Manager1 : get SPID and IAS credentials
-  note right: from local peer config
-
-  Manager1 -> ECC1 ++: create (SPID)
-  note left: this is a cc2cc call
-
-
-  ECC1 -> TLCC1 ++: getInfo
-  note over TLCC1
-    This information could also be retrieved elsewhere if easier. At this point in time
-    it does not have to be authenticated (and is anyway passed through untrusted ECC),
-    consistency will be verified later in registration (Remember that TLCC_MR is defined
-    and agreed by human consensus in the ERCC chaincode definition)
-  end note
-  return TLCC_MR, CH_id
-
-
-  |||
-
-  create participant "ECC_Enclave" as Enclave1 order 30 #99FF99
-  ECC1 -> Enclave1 ++: create (SPID, CC_NAME, CC_VER, Org_1, TLCC_MR, CH_id)
-  note left
-    this is a cgo call
-    Org_1 denotes the peer's MSP_ID
-  end note
+  Admin -> ECC1 ++: invoke create(SPID)
+
+  'ECC1 -> TLCC1 ++: getInfo
+  'note over TLCC1
+  '  This information could also be retrieved elsewhere if easier. At this point in time
+  '  it does not have to be authenticated (and is anyway passed through untrusted ECC),
+  '  consistency will be verified later in registration (Remember that TLCC_MR is defined
+  '  and agreed by human consensus in the ERCC chaincode definition)
+  'end note
+  'return TLCC_MR, CH_id
+
+  ECC1 -> Enclave1 ++: invoke create(SPID)
+
+  Enclave1 -> ECC1 ++: get_signed_proposal
+  return Signed_Proposal
+  Enclave1 -> Enclave1 : CC_Id, CC_Version <- extract_cc_info(Signed_Proposal) 
+  Enclave1 -> ECC1 ++: get_mspid
+  return MSPID
+  Enclave1 -> ECC1 ++: get_channel_id
+  return Channel_Id
+  group fpc-cc-tlcc binding (to be discussed)
+  Enclave1 -> ERCC1 ++: get_tlcc_mrenclave()
+  return TLCC_MREnclave
+  end group
 
   Enclave1 -> Enclave1 : keyGen
   note left
-    keyGen produces the following keys:
-    ECC_S_PK : Sig pub key
-    ECC_S_SK : Sig priv key
-    ECC_E_PK : Enc pub key
-    ECC_E_SK : Enc priv key
-  end note
-
-  note left Enclave1
-    ECC_ID is //short// unique identifier
-    ECC_ID = Hash(ECC_S_PK)
+    keyGen generates the keys stored in the key ring,
+    and described in the Credential protobuf
   end note
 
-  Enclave1  -> ECC1 ++: binding (ECC_S_PK, ECC_E_PK)
+  group Org-Enclave binding (to be discussed)
+  Enclave1  -> ECC1 ++: binding (ECC_VK)
```

Comment:
any reason you dropped the encryption key from the binding?  Also i would leave what is exactly signed (which should be more than only ECC_VK but also context). I haven't checked, if you added a corresponding data struct to the other diagram, just a reference to that datastructure should do as well (e.g., simular to what you have done for Credential)

Reply:
no reason, I tagged this as TBD.
I am actually not sure anymore about this step. This is for the Org to certify (i.e., taking accountability of) the enclave, and then let the enclave attest that. Do we still need this?
The certification can be done peer-side, then the certificate check is up to TLCC.

Reply:
I would keep it as it was unless you can argue that we don't need that binding anymore.